libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/viewmodel/ContainerVisibilityViewModel.kt

```diff
@@ -0,0 +1,92 @@
+package com.mapbox.navigation.dropin.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.mapbox.navigation.dropin.contract.ContainerVisibilityAction
+import com.mapbox.navigation.dropin.contract.ContainerVisibilityResult
+import com.mapbox.navigation.dropin.contract.toProcessor
+import com.mapbox.navigation.dropin.contract.toResult
+import com.mapbox.navigation.dropin.state.ContainerVisibilityState
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.flow.scan
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+
+@ExperimentalCoroutinesApi
+class ContainerVisibilityViewModel : ViewModel() {
+    private val _containerVisibilityState: MutableStateFlow<ContainerVisibilityState> =
```

Comment:
This duplicates the `_navigationState: MutableStateFlow<NavigationState>` in the `NavigationStateViewModel`

In reality, this is a transformation of the `Flow<NavigationState>`. The ViewModel here appears to be overkill

Reply:
I agree and as explained above will fix this in a separate PR so as to unblock this.

Reply:
I agree that additional view model might be an overkill. The visibility value will ultimately be a transformation of state, visibility settings from config, and data availability.